### PR TITLE
Integrate p2p feat/refactor_net_messenger

### PR DESCRIPTION
**Pull request by sstanculeanu**
## Reasoning behind the pull request
- refactoring
  
## Proposed changes
- refactor network messenger

## Testing procedure
- half-half with `rc/v1.5.0`

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: process/txsSender/txsSender.go [hunk 1]
```diff
@@ -173,18 +173,15 @@ func (ts *txsSender) sendBulkTransactionsFromShard(transactions [][]byte, sender
...
-				"size", len(bufferToSend),
-			)
-			err = ts.networkMessenger.BroadcastOnChannelBlocking(
-				SendTransactionsPipe,
-				identifier,
-				bufferToSend,
-			)
-			log.LogIfError(err)
-		}(buff)
+		log.Trace("txsSender.sendBulkTransactionsFromShard",
```
Comment (iulianpascalau) [inline process/txsSender/txsSender.go:176]: :+1:

**Review by iulianpascalau — APPROVED** [review]

**Review by ssd04 — DISMISSED** [review]

**Review by gabi-vuls — DISMISSED** [review]
System test passed

File: go.mod [hunk 2]
```diff
@@ -18,7 +18,7 @@ require (
 	github.com/multiversx/mx-chain-crypto-go v1.2.5
 	github.com/multiversx/mx-chain-es-indexer-go v1.3.8
 	github.com/multiversx/mx-chain-logger-go v1.0.11
-	github.com/multiversx/mx-chain-p2p-go v1.0.10
+	github.com/multiversx/mx-chain-p2p-go v1.0.11-0.20230126102814-52bb70d19905
```
[the remark this answers is not in the document]
Comment (sstanculeanu) [reply go.mod]: updated with latest `rc/v1.5.0` commit

**Review by gabi-vuls — APPROVED** [review]

**Review by ssd04 — DISMISSED** [review]

File: go.mod [hunk 3]
```diff
@@ -18,7 +18,7 @@ require (
 	github.com/multiversx/mx-chain-crypto-go v1.2.5
 	github.com/multiversx/mx-chain-es-indexer-go v1.3.8
 	github.com/multiversx/mx-chain-logger-go v1.0.11
-	github.com/multiversx/mx-chain-p2p-go v1.0.11-0.20230126102814-52bb70d19905
+	github.com/multiversx/mx-chain-p2p-go v1.0.11-0.20230130113204-00e2c90306f3
```
Comment (iulianpascalau) [inline go.mod:21]: proper release

Comment (sstanculeanu) [reply go.mod]: pushed

**Review by iulianpascalau — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]

**Review by ssd04 — APPROVED** [review]

**Pull request merged**